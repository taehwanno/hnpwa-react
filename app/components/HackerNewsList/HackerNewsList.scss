.HackerNewsList {
  padding-left: 20px;

  > li {
    list-style: none;
  }
}

.HackerNewsList__noti {
  text-align: center;
}

.HackerNewsList__item {
  margin: 30px 0;
}

.HackerNewsList__index {
  font-size: 2em;
  position: absolute;
  text-align: center;
  width: 1em;
}
