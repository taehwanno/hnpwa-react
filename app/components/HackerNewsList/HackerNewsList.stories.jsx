import Immutable from 'immutable';
import React from 'react';
import { storiesOf } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

import HackerNewsList from './';

const stories = storiesOf('HackerNewsList', module);

stories
  .addDecorator(story => (
    <MemoryRouter initialEntries={['/news/1']}>
      {story()}
    </MemoryRouter>
  ))
  .add('fetching data', () => <HackerNewsList isFetching />)
  .add('no data', () => <HackerNewsList feeds={Immutable.List()} isFetching={false} />)
  .add('with data', () => (
    <HackerNewsList
      feeds={Immutable.fromJS([
        {
          id: 15350263,
          title: 'Keybase\'s mission is to make encryption mainstream',
          points: 172,
          user: 'BradyDale',
          time: 1506534023,
          timeAgo: '20 hours ago',
          commentsCount: 79,
          type: 'link',
          url: 'http://observer.com/2017/09/keybase-max-krohn-chris-coyne-okcupid/',
          domain: 'observer.com',
        },
        {
          id: 15351433,
          title: 'What happens after a defendant is found not guilty by reason of insanity?',
          points: 74,
          user: 'mcone',
          time: 1506540664,
          timeAgo: '18 hours ago',
          commentsCount: 131,
          type: 'link',
          url: 'https://www.nytimes.com/2017/09/27/magazine/when-not-guilty-is-a-life-sentence.html',
          domain: 'nytimes.com',
        },
        {
          id: 15334928,
          title: 'Hunt and House: Depictions of Medieval Life in German Playing Cards (2016)',
          points: 21,
          user: 'benbreen',
          time: 1506381168,
          timeAgo: '3 days ago',
          commentsCount: 5,
          type: 'link',
          url: 'http://www.metmuseum.org/blogs/in-season/2016/hunt-and-house',
          domain: 'metmuseum.org',
        },
        {
          id: 15350264,
          title: 'Keybase\'s mission is to make encryption mainstream',
          points: 172,
          user: 'BradyDale',
          time: 1506534023,
          timeAgo: '20 hours ago',
          commentsCount: 79,
          type: 'link',
          url: 'http://observer.com/2017/09/keybase-max-krohn-chris-coyne-okcupid/',
          domain: 'observer.com',
        },
        {
          id: 15351434,
          title: 'What happens after a defendant is found not guilty by reason of insanity?',
          points: 74,
          user: 'mcone',
          time: 1506540664,
          timeAgo: '18 hours ago',
          commentsCount: 131,
          type: 'link',
          url: 'https://www.nytimes.com/2017/09/27/magazine/when-not-guilty-is-a-life-sentence.html',
          domain: 'nytimes.com',
        },
        {
          id: 15334924,
          title: 'Hunt and House: Depictions of Medieval Life in German Playing Cards (2016)',
          points: 21,
          user: 'benbreen',
          time: 1506381168,
          timeAgo: '3 days ago',
          commentsCount: 5,
          type: 'link',
          url: 'http://www.metmuseum.org/blogs/in-season/2016/hunt-and-house',
          domain: 'metmuseum.org',
        },
        {
          id: 15350265,
          title: 'Keybase\'s mission is to make encryption mainstream',
          points: 172,
          user: 'BradyDale',
          time: 1506534023,
          timeAgo: '20 hours ago',
          commentsCount: 79,
          type: 'link',
          url: 'http://observer.com/2017/09/keybase-max-krohn-chris-coyne-okcupid/',
          domain: 'observer.com',
        },
        {
          id: 15351435,
          title: 'What happens after a defendant is found not guilty by reason of insanity?',
          points: 74,
          user: 'mcone',
          time: 1506540664,
          timeAgo: '18 hours ago',
          commentsCount: 131,
          type: 'link',
          url: 'https://www.nytimes.com/2017/09/27/magazine/when-not-guilty-is-a-life-sentence.html',
          domain: 'nytimes.com',
        },
        {
          id: 15334925,
          title: 'Hunt and House: Depictions of Medieval Life in German Playing Cards (2016)',
          points: 21,
          user: 'benbreen',
          time: 1506381168,
          timeAgo: '3 days ago',
          commentsCount: 5,
          type: 'link',
          url: 'http://www.metmuseum.org/blogs/in-season/2016/hunt-and-house',
          domain: 'metmuseum.org',
        },
        {
          id: 15350262,
          title: 'Keybase\'s mission is to make encryption mainstream',
          points: 172,
          user: 'BradyDale',
          time: 1506534023,
          timeAgo: '20 hours ago',
          commentsCount: 79,
          type: 'link',
          url: 'http://observer.com/2017/09/keybase-max-krohn-chris-coyne-okcupid/',
          domain: 'observer.com',
        },
        {
          id: 15351423,
          title: 'What happens after a defendant is found not guilty by reason of insanity?',
          points: 74,
          user: 'mcone',
          time: 1506540664,
          timeAgo: '18 hours ago',
          commentsCount: 131,
          type: 'link',
          url: 'https://www.nytimes.com/2017/09/27/magazine/when-not-guilty-is-a-life-sentence.html',
          domain: 'nytimes.com',
        },
        {
          id: 15334922,
          title: 'Hunt and House: Depictions of Medieval Life in German Playing Cards (2016)',
          points: 21,
          user: 'benbreen',
          time: 1506381168,
          timeAgo: '3 days ago',
          commentsCount: 5,
          type: 'link',
          url: 'http://www.metmuseum.org/blogs/in-season/2016/hunt-and-house',
          domain: 'metmuseum.org',
        },
      ])}
    />
  ));
