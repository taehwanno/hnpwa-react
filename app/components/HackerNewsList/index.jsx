import Immutable from 'immutable';
import React from 'react';
import PropTypes from 'prop-types';

import HackerNewsListItem from 'components/HackerNewsListItem';
import LoadingIndicator from 'components/LoadingIndicator';

import './HackerNewsList.scss';

const propTypes = {
  feeds: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  isFetching: PropTypes.bool,
};

const defaultProps = {
  feeds: Immutable.List(),
  isFetching: false,
};

function HackerNewsList({ feeds, isFetching }) {
  const haveNoItems = !isFetching && feeds.size === 0;

  return (
    <ul className="HackerNewsList">
      <LoadingIndicator
        active={isFetching}
        style={{ position: 'absolute', left: 'calc(50% - 24px)', marginTop: '10px' }}
      />
      {haveNoItems && <p className="HackerNewsList__noti">There are no items to show.</p>}
      {!isFetching && feeds.map((feed, index) => (
        <li className="HackerNewsList__item" key={feed.get('id')}>
          <span className="HackerNewsList__index">{index + 1}</span>
          <HackerNewsListItem {...feed.toJS()} />
        </li>
      )).toArray()}
    </ul>
  );
}

HackerNewsList.propTypes = propTypes;
HackerNewsList.defaultProps = defaultProps;

export default HackerNewsList;
