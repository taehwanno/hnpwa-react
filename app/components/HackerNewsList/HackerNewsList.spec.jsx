import Immutable from 'immutable';
import React from 'react';
import { shallow } from 'enzyme';

import HackerNewsList from './';

describe('<HackerNewsList />', () => {
  it('should match snapshot when render default', () => {
    const wrapper = shallow(<HackerNewsList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when render feeds', () => {
    const feeds = Immutable.fromJS([
      {
        id: 15350263,
        title: 'Keybase\'s mission is to make encryption mainstream',
        points: 172,
        user: 'BradyDale',
        time: 1506534023,
        timeAgo: '20 hours ago',
        commentsCount: 79,
        type: 'link',
        url: 'http://observer.com/2017/09/keybase-max-krohn-chris-coyne-okcupid/',
        domain: 'observer.com',
      },
      {
        id: 15351433,
        title: 'What happens after a defendant is found not guilty by reason of insanity?',
        points: 74,
        user: 'mcone',
        time: 1506540664,
        timeAgo: '18 hours ago',
        commentsCount: 131,
        type: 'link',
        url: 'https://www.nytimes.com/2017/09/27/magazine/when-not-guilty-is-a-life-sentence.html',
        domain: 'nytimes.com',
      },
    ]);
    const wrapper = shallow(<HackerNewsList feeds={feeds} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have .HackerNewsList__noti when have no items', () => {
    const wrapper = shallow(<HackerNewsList feeds={Immutable.List()} isFetching={false} />);
    expect(wrapper.find('.HackerNewsList__noti')).toHaveLength(1);
  });
});
