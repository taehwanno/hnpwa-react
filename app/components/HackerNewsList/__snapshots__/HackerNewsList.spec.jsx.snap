// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HackerNewsList /> should match snapshot when render default 1`] = `
<ul
  className="HackerNewsList"
>
  <LoadingIndicator
    active={false}
    style={
      Object {
        "left": "calc(50% - 24px)",
        "marginTop": "10px",
        "position": "absolute",
      }
    }
  />
  <p
    className="HackerNewsList__noti"
  >
    There are no items to show.
  </p>
</ul>
`;

exports[`<HackerNewsList /> should match snapshot when render feeds 1`] = `
<ul
  className="HackerNewsList"
>
  <LoadingIndicator
    active={false}
    style={
      Object {
        "left": "calc(50% - 24px)",
        "marginTop": "10px",
        "position": "absolute",
      }
    }
  />
  <li
    className="HackerNewsList__item"
    key="15350263"
  >
    <span
      className="HackerNewsList__index"
    >
      1
    </span>
    <HackerNewsListItem
      commentsCount={79}
      domain="observer.com"
      id={15350263}
      points={172}
      time={1506534023}
      timeAgo="20 hours ago"
      title="Keybase's mission is to make encryption mainstream"
      type="link"
      url="http://observer.com/2017/09/keybase-max-krohn-chris-coyne-okcupid/"
      user="BradyDale"
    />
  </li>
  <li
    className="HackerNewsList__item"
    key="15351433"
  >
    <span
      className="HackerNewsList__index"
    >
      2
    </span>
    <HackerNewsListItem
      commentsCount={131}
      domain="nytimes.com"
      id={15351433}
      points={74}
      time={1506540664}
      timeAgo="18 hours ago"
      title="What happens after a defendant is found not guilty by reason of insanity?"
      type="link"
      url="https://www.nytimes.com/2017/09/27/magazine/when-not-guilty-is-a-life-sentence.html"
      user="mcone"
    />
  </li>
</ul>
`;
